unlearnfeature.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+// Completely untested, largely stolen from 'learnfeature'
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"gopkg.in/redis.v5"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// UnlearnFeature allows crbot to unlearn existing calls
+type UnlearnFeature struct {
+	featureRegistry *FeatureRegistry
+	redisClient     *redis.Client
+}
+
+// NewUnlearnFeature returns a new UnlearnFeature.
+func NewUnlearnFeature(featureRegistry *FeatureRegistry, redisClient *redis.Client) *UnlearnFeature {
+	return &UnlearnFeature{
+		featureRegistry: featureRegistry,
+		redisClient:     redisClient,
+	}
+}
+
+// GetName returns the named type of this feature.
+func (f *UnlearnFeature) GetName() string {
+	return Name_Unlearn
+}
+
+// GetType returns the type of this feature.
+func (f *UnlearnFeature) GetType() int {
+	return Type_Unlearn
+}
+
+// Invokable returns whether the user can execute this command by name.
+func (f *UnlearnFeature) Invokable() bool {
+	return true
+}
+
+// Parse parses the given unlearn command.
+func (f *UnlearnFeature) Parse(splitContent []string) (*Command, error) {
+	if splitContent[0] != f.GetName() {
+		fatal("parseUnlearn called with non-unlearn command", errors.New("wat"))
+	}
+
+	callRegexp := regexp.MustCompile("(?s)^[[:alnum:]].*$")
+	responseRegexp := regexp.MustCompile("(?s)^[^/?!].*$")
```

Comment:
There's no response to parse this against. I think this will crash (splitcontent should only have 2 elements)